shell/platform/fuchsia/flutter/pointer_injector_delegate.cc

```diff
@@ -0,0 +1,318 @@
+// Copyright 2022 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "pointer_injector_delegate.h"
+#include "flutter/fml/logging.h"
+
+namespace flutter_runner {
+
+using fup_Config = fuchsia::ui::pointerinjector::Config;
+using fup_Context = fuchsia::ui::pointerinjector::Context;
+using fup_Data = fuchsia::ui::pointerinjector::Data;
+using fup_DeviceType = fuchsia::ui::pointerinjector::DeviceType;
+using fup_DispatchPolicy = fuchsia::ui::pointerinjector::DispatchPolicy;
+using fup_Event = fuchsia::ui::pointerinjector::Event;
+using fup_EventPhase = fuchsia::ui::pointerinjector::EventPhase;
+using fup_PointerSample = fuchsia::ui::pointerinjector::PointerSample;
+using fup_Target = fuchsia::ui::pointerinjector::Target;
+using fup_Viewport = fuchsia::ui::pointerinjector::Viewport;
+using fuv_ViewRef = fuchsia::ui::views::ViewRef;
+const auto fup_MAX_INJECT = fuchsia::ui::pointerinjector::MAX_INJECT;
+
+namespace {
+
+// clang-format off
+  static constexpr std::array<float, 9> kIdentityMatrix = {
+    1, 0, 0, // column one
+    0, 1, 0, // column two
+    0, 0, 1, // column three
+  };
+// clang-format on
+
+}  // namespace
+
+bool PointerInjectorDelegate::HandlePlatformMessage(
+    rapidjson::Value request,
+    fml::RefPtr<flutter::PlatformMessageResponse> response) {
+  if (!registry_->is_bound()) {
+    FML_LOG(WARNING)
+        << "Lost connection to fuchsia.ui.pointerinjector.Registry";
+    return false;
+  }
+
```

Comment:
We need to wait here for the ViewRef to be valid, and drop any events if it is not instead of buffering them (per Jaeheon's advice)

Also, if a View was previously valid but has become invalid (because the view was destroyed or its process terminated), we need to track that in here and drop the event in that case (instead of buffering it forever below).  As a consequence, the dart code in fuchsia_views must never re-use a viewId for new views

Reply:
Since we no longer require to buffer any pointer events n the case the view is not created, this problem is resolved. We will only dispatch the pointer events if the view is present in |valid_views_|. When the view is destroyed/disconnected its entry is removed from |valid_views_| and therefore all the pointer events will be dropped.